src/hf.f90

```diff
@@ -0,0 +1,318 @@
+module hf
+   use const
+
+   implicit none
+
+   type diis_t
+      ! Contains matrices for use in the DIIS procedure
+      logical :: use_diis = .true.
+      integer :: n_errmat = 6
+      integer :: n_active = 0
+      integer :: iter = 0
+      real(p), allocatable :: e(:,:,:)

```

Comment:
definitely document what these are - also helpful to loosely know what the order of the subscripts is eg:
!F(norb,norb,n_max_diis_steps)